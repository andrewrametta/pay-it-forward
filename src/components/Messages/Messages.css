.messages-wrapper {
  width: 100%;
  padding: 20px;
  background: lightgray;
  display: flex;
  flex-direction: row;
}

.conversation-list {
  height: 500px;
  overflow: scroll;
  flex: 1;
  background: coral;
}

.chat-messages-box {
  flex: 3;
}
