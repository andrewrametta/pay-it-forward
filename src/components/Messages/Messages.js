import React, { useEffect, useState, useContext } from "react";
import AppContext from "../../AppContext";
import AuthAPIService from "../../services/auth-api-service";
import "./Messages.css";
import { Link, Route } from "react-router-dom";
import Chat from "../Chat/Chat";

export default function Messages(props) {
  const [error, setError] = useState("");
  const { type, conversations, setConversations, setChatOn } = useContext(
    AppContext
  );

  useEffect(() => {
    setChatOn(true);
    AuthAPIService.getConversation()
      .then((conversations) => {
        setConversations(conversations);
      })
      .catch((error) => {
        setError(error);
      });
  }, [setConversations, setError, setChatOn]);

  return (
    <div className="messages-wrapper">
      {error && <h2>error</h2>}
      <section className="conversation-list">
        {conversations.length > 0 ? (
          conversations.map((conversation, indx) => (
            <Link key={indx} to={`/messages/${conversation.id}`}>
              {type === "org" ? (
                <h2>{conversation.username2}</h2>
              ) : (
                <h2>{conversation.username}</h2>
              )}
              <p>{conversation.title}</p>
            </Link>
          ))
        ) : (
          <h3>Looks like there are no Messages at this time</h3>
        )}
      </section>
      <section className="chat-box">
        <Route exact path="/messages/:conversation_id" component={Chat} />
      </section>
    </div>
  );
}
