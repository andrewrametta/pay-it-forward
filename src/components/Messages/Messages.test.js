import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Messages from "./Messages";
// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import AppContext from "../../AppContext";

configure({ adapter: new Adapter() });

describe(`Messages component`, () => {
  it.only("renders the complete page", () => {
    const wrapper = shallow(
      <AppContext.Provider>
        <Messages />
      </AppContext.Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
