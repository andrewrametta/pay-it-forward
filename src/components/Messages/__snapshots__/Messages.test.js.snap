// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages component renders the complete page 1`] = `<Messages />`;
